smarts/sstudio/generators.py

```diff
@@ -171,10 +160,19 @@ def plan_and_save(
                 self._log.info(f"Routes at routes={route_path} already exist, skipping")
                 return None
 
+        if traffic.engine == "SMARTS":
+            self._writexml(traffic, True, route_path)
+            return route_path
```

Comment:
No, we're not using Sumo's duarouter for Smarts traffic.  Indeed, that's the point of this "short circuit" path (i.e., the `return` on line 165), to skip the duarouter stage that comes next within the function (around line 183).

Instead, for Smarts traffic, we are just writing the xml file that would have been _used_ _as_ _input_ by the duarouter.  The `LocalTrafficProvider` can read that and do its routing from that alone.  

So SUMO doesn't have to be present at all if you only want Smarts traffic.

_However_... I should point out that I was considering implementing _something_ _like_ duarouter for Smarts traffic (cf. the comment on lines 312-313 of `generators.py`) and/or even making it so `LocalTrafficProvider` could _also_ take duarouter-generate files (`*.rou.xml`) as input to both support cross-compatibility as well as dynamic routing.   But I ultimately held off on this because I haven't yet worked through all of the performance implications of dynamic routing.

But even if I did something like this (in a future PR), I would maintain the ability to have Smarts traffic without requiring Sumo.

Reply:
This discussion made me realize that the import for Sumo's duarouter should come _after_ the Smarts short-circuit path, so I just changed that.  Thanks!